### PR TITLE
Add --test option to run_tests command

### PR DESCRIPTION
**Pull request by troydai**
Address issue #3885 
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [ ] Each command and parameter has a meaningful description.
- [ ] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

File: scripts/automation/tests/run.py [hunk 1]
```diff
@@ -7,12 +7,40 @@
...
 
 
-def run_tests(modules, parallel, run_live):
+def get_unittest_runner(tests):
+    test_cases = list(tests)
+
+    def _runner(module_paths):
+        from subprocess import check_call, CalledProcessError
+        if len(module_paths) > 1:
+            print('When --test is given, no more than 1 module and be selected.')
```
Comment (tjprescott) [inline scripts/automation/tests/run.py]: typo: can be selected

File: scripts/automation/tests/run.py [hunk 2]
```diff
@@ -23,29 +51,35 @@ def run_tests(modules, parallel, run_live):
...
     parse = argparse.ArgumentParser('Test tools')
     parse.add_argument('--module', dest='modules', action='append',
-                       help='The modules of which the test to be run. Accept short names, except '
-                            'azure-cli, azure-cli-core and azure-cli-nspkg. The modules list can '
-                            'also be set through environment variable AZURE_CLI_TEST_MODULES. The '
-                            'value should be a string of comma separated module names. The '
-                            'environment variable will be overwritten by command line parameters.')
+                       help='The modules of which the test to be run. Accept short names, except azure-cli, '
+                            'azure-cli-core and azure-cli-nspkg. The modules list can also be set through environment '
+                            'variable AZURE_CLI_TEST_MODULES. The value should be a string of comma separated module '
```
Comment (tjprescott) [inline scripts/automation/tests/run.py]: A little strange--never knew this. Why not make it space separated using the native argparse supported nargs='+'?

File: scripts/automation/tests/run.py [hunk 3]
```diff
@@ -23,29 +51,35 @@ def run_tests(modules, parallel, run_live):
...
+                            'names. The environment variable will be overwritten by command line parameters.')
     parse.add_argument('--parallel', action='store_true',
```
Comment (tjprescott) [inline scripts/automation/tests/run.py]: We'd talked about making parallel the default for nosetests.  Is it more complicated that tinkering with this?

File: scripts/automation/tests/run.py (hunk 2 above)
Comment (troydai) [inline scripts/automation/tests/run.py]: Yeah, I can do that.

File: scripts/automation/tests/run.py (hunk 3 above)
Comment (troydai) [inline scripts/automation/tests/run.py]: It's not. I was thinking not to do too many changes in one PR.

**Comment by tjprescott** [comment]
@troydai can we get the changes made here and get this in?

**Comment by troydai** [comment]
Will do soon.

**Comment by troydai** [comment]
Please take another look. @tjprescott 
I'll flip the non-parallel option later.

File: scripts/automation/tests/run.py [hunk 4]
```diff
@@ -68,7 +68,7 @@ def run_tests(modules, parallel, run_live, tests):
 
 if __name__ == '__main__':
     parse = argparse.ArgumentParser('Test tools')
-    parse.add_argument('--module', dest='modules', action='append',
+    parse.add_argument('--module', dest='modules', nargs='+',
                        help='The modules of which the test to be run. Accept short names, except azure-cli, '
                             'azure-cli-core and azure-cli-nspkg. The modules list can also be set through environment '
                             'variable AZURE_CLI_TEST_MODULES. The value should be a string of comma separated module '
```
Comment (tjprescott) [inline scripts/automation/tests/run.py]: With nargs='+' this would be a space-separated list instead of comma-separated.

**Comment by troydai** [comment]
Will fix the message and push a merge.

**Pull request merged**